src/ICeWall/src/ICeWall.tcl

```
@@ -3561,26 +3904,1336 @@ namespace eval ICeWall {
+    } else {
+      utl::error PAD 215 "The number of pads within a bump pitch has not been specified"
+    }
+  }
+
+  proc add_pad {args} {
+    variable footprint
+    variable block
+    variable type_index
+    variable db
+    variable tech
+
+    set db [::ord::get_db]
+    set tech [$db getTech]
+    set block [ord::get_db_block]
+
+    set padcell_name {}
+
+    if {[llength $args] % 2 == 1} {
+        utl::error PAD 109 "Incorrect number of arguments for add_pad - expected an even number, received [llength $args]"
+    }
+
+    set padcell {}
+    # Determine the padcell name
+    if {[dict exists $args -name]} {
+      set padcell_name [dict get $args -name]
+    } else {
+      if {[dict exists $args -type]} {
+        set type [dict get $args -type]
+        if {![dict exists $type_index $type]} {
+          dict set type_index $type 0
+        } 
+        set idx [dict get $type_index $type]
+        set padcell_name "${type}_$idx"
+        incr idx
+        dict set type_index $type $idx
+      } else {
+        utl::error PAD 110 "Must specify -type option if -name is not spedified"
+      }
+    }
+    if {[dict exists $footprint padcell $padcell_name]} {
+      utl::error PAD 216 "A padcell with the name $padcell_name already exists"
+    }
+    dict set padcell name $padcell_name
+
+    set process_args $args
+    while {[llength $process_args] > 0} {
+      set arg [lindex $process_args 0]
+      set value [lindex $process_args 1]
+
+      switch $arg {
+        -name      {dict set padcell name $value}
+        -signal    {dict set padcell signal_name $value}
+        -edge      {dict set padcell side [check_edge_name $value]}
+        -type      {dict set padcell type [check_cell_type $value]}
+        -cell      {dict set padcell cell_name [check_cell_name $value]}
+        -location  {dict set padcell cell [check_location $value]}
+        -bump      {dict set padcell bump [check_bump $value]}
```

This should report an error if the same bump is used more than once.

---

Added consistency check to ensure that the same row/col combination is only used once